test/cloudtest/pkg/providers/packet/packet_cleanup/tool.go

```diff
@@ -0,0 +1,188 @@
+
+	rootCmd.Flags().StringVar(&rootCmd.cmdArguments.sshPrefix, "ssh-prefix", "dev-ci-cloud", "SSH key prefix to delete")
+	rootCmd.Flags().StringArrayVar(&rootCmd.cmdArguments.clusterPrefix, "cluster-prefix", []string{"Worker-packet-", "Master-packet-"}, "SSH key prefix to delete")
+
+	var versionCmd = &cobra.Command{
+		Use:   "version",
+		Short: "Print the version number of packet cleanup tool",
+		Long:  `All software has versions.`,
+		Run: func(cmd *cobra.Command, args []string) {
+			fmt.Println("Packet cleanup tool -- HEAD")
+		},
+	}
+	rootCmd.AddCommand(versionCmd)
+}
+
+func main() {
+	logrus.Infof("Packet cleanup tool.")
+
+	var rootCmd = &packetCleanupCmd{
+		cmdArguments: &Arguments{},
+	}
+	rootCmd.Use = "packet_cleanup"
+	rootCmd.Short = "Packet cleanup tool"
+	rootCmd.Long = `Allow to cleanup packet instance from ssh keys and old clusters`
+	rootCmd.Run = func(cmd *cobra.Command, args []string) {
+		doCleanup(rootCmd)
+	}
+	rootCmd.Args = func(cmd *cobra.Command, args []string) error {
+		return nil
+	}
+
+	initCmd(rootCmd)
+
+	if err := rootCmd.Execute(); err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+}
+
+func doCleanup(cmd *packetCleanupCmd) {
+	if cmd.cmdArguments.projectID == "" || cmd.cmdArguments.token == "" {
+		logrus.Errorf("Please specify both projectID and token")
+		os.Exit(1)
+	}
+
+	helper, err := packethelper.NewPacketHelper(cmd.cmdArguments.projectID, cmd.cmdArguments.token)
+	if err != nil {
+		logrus.Errorf("Error accessing packet %v", err)
+		os.Exit(1)
+	}
+	if cmd.cmdArguments.deleteSSHKeys {
+		deleteSSHKeys(cmd, helper)
+	}
+	if cmd.cmdArguments.deleteClusters {
+		deleteClusters(cmd, helper)
+	}
+}
+
+func deleteClusters(cmd *packetCleanupCmd, helper *packethelper.PacketHelper) {
+	pageNumber := 0
+	for {
+		t1 := time.Now()
+		devs, _, err := helper.Client.Devices.List(helper.Project.ID, &packngo.ListOptions{
+			Page:    pageNumber,
+			PerPage: 10,
+		})
+		if len(devs) == 0 {
+			break
+		}
+		logrus.Infof("Retrieve list of clusters %v", time.Since(t1))
+		if err != nil {
+			logrus.Errorf("Failed to retrieve list of clusters %v", err)
+			break
+		}
+		for i := 0; i < len(devs); i++ {
+			d := &devs[i]
+			keyCreateTime, err := time.Parse(time.RFC3339, d.Created)
+			if err != nil {
+				logrus.Errorf("Failed to parse time %v", err)
+				continue
+			}
+			sinceValue := time.Since(keyCreateTime)
+			logrus.Infof("Checking cluster %v uptime: %v, state: %v", d.Hostname, sinceValue, d.State)
+			if checkPrefix(d.Hostname, cmd.cmdArguments.clusterPrefix) {
+				if sinceValue > cmd.cmdArguments.clusterLifetime {
+					logrus.Infof("-----> Device is marked for deletion.")
+					_, errd := helper.Client.Devices.Delete(d.ID)
+					if errd != nil {
+						logrus.Errorf("Error during delete %v", errd)
+					}
+				}
+			}
+		}
+		pageNumber++
+	}
+	logrus.Infof("All clusters fit into timeframe %v", cmd.cmdArguments.clusterLifetime)
```

Comment:
It might be nice to have the number of clusters deleted and remaining logged here